### PR TITLE
test: Forge::modifyColumn() and null

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #7301~~

**Description**
From https://zenn.dev/naente/articles/50935eb55c14ea (CodeIgniter4 PostgreSQL always adds NOT NULL constraints when running `modifyColumn()`)

See the current behavior: https://github.com/codeigniter4/CodeIgniter4/pull/7302#issuecomment-1444980360

- add test for `Forge::modifyColumn()` and null

**This bug was fixed in v4.3.4**:
-  #7371

Related #7235

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
@codeigniter4/database-team 
It seems Postgre, SQLSRV, SQLite3 add `NOT NULL` constraints when running `modifyColumn()`.
See the test: https://github.com/codeigniter4/CodeIgniter4/pull/7302/files#diff-489cc58f8341d1fa6ca26b7bb857562d4274924d2dc691ca64e9dc609731b8eaR1285
Are these bugs or not?

**Comment by sclubricants** [comment]
Looks like nullable is the default. What is the matter with this?

**Comment by kenjis** [comment]
I did not look into all drivers, but Postgre changes to NULL in all cases.
There is no way to keep nuallble. So it is clearly a bug.

Another issue is this behavior is not consistent. As you see all tests pass
on MySQLi and OCI8.

If a dev changes DB driver, e.g. MySQLi to Postgres,
the definitiion of the table will change when refreshing migrations.

**Comment by kenjis** [comment]
```diff
--- a/tests/system/Database/Live/ForgeTest.php
+++ b/tests/system/Database/Live/ForgeTest.php
@@ -1301,6 +1301,9 @@ final class ForgeTest extends CIUnitTestCase
 
         $this->db->resetDataCache();
 
+        $fields = $this->db->getFieldData('forge_test_modify');
+        dd($fields);
+
         $col1 = $this->getMetaData('col1');
         $this->assertTrue($col1->nullable);
         $col2 = $this->getMetaData('col2');
```
Postgres:
```
array (3) [
    0 => stdClass#249 (5) (
        public 'name' -> string (4) "col1"
        public 'type' -> string (17) "character varying"
        public 'nullable' -> boolean false
        public 'default' -> null
        public 'max_length' -> string (1) "1"
    )
    1 => stdClass#74 (5) (
        public 'name' -> string (4) "col2"
        public 'type' -> string (17) "character varying"
        public 'nullable' -> boolean false
        public 'default' -> null
        public 'max_length' -> string (1) "1"
    )
    2 => stdClass#75 (5) (
        public 'name' -> string (4) "col3"
        public 'type' -> string (17) "character varying"
        public 'nullable' -> boolean false
        public 'default' -> null
        public 'max_length' -> string (1) "1"
    )
]
```

**Comment by michalsn** [comment]
Seems like a bug to me.

**Comment by kenjis** [comment]
Rebased and now this PR removes only unneeded test code.

**Review by michalsn — APPROVED** [review]

**Comment by MGatner** [comment]
Looks good, thanks!
